evap/staff/views.py

```diff
@@ -479,17 +426,59 @@ def course_comments(request, semester_id, course_id):
     semester = get_object_or_404(Semester, id=semester_id)
     course = get_object_or_404(Course, id=course_id)
 
-    textanswers = course.textanswer_set.filter(checked=True)
-
-    textanswers_by_question = []
-    for question_id in textanswers.values_list("question", flat=True).distinct():
-        textanswers_by_question.append((Question.objects.get(id=question_id), textanswers.filter(question=question_id)))
+    course_sections = []
+    contributor_sections = []
+    for questionnaire, contribution in questionnaires_and_contributions(course):
+        comments = []
+        for question in questionnaire.question_set.all():
+            if question.is_text_question:
+                answers = get_filtered_answers(course, contribution, question, exclude_text_answer_states=[])
+                if answers:
+                    comments.append(TextResult(
+                        question=question,
+                        answers=answers,
+                    ))
+        if not comments:
+            continue
+        elif contribution.is_general:
+            course_sections.append(CommentSection(questionnaire, contribution.contributor, False, comments))
+        else:
+            contributor_sections.append(CommentSection(questionnaire, contribution.contributor, contribution.responsible, comments))
 
-    template_data = dict(semester=semester, course=course, textanswers_by_question=textanswers_by_question)
+    template_data = dict(semester=semester, course=course, course_sections=course_sections, contributor_sections=contributor_sections)
     return render(request, "staff_course_comments.html", template_data)
 
 
 @staff_required
+def course_comments_update_publish(request):
+    comment_id = request.POST["id"]
+    publish_status = request.POST["status"]
+    course_id = request.POST["course_id"]
+
+    course = Course.objects.get(pk=course_id)
+    answer = TextAnswer.objects.get(pk=comment_id)
+
+    if publish_status == 'publish':
+        answer.publish()
+    elif publish_status == 'private':
+        answer.make_private()
+    elif publish_status == 'hide':
+        answer.hide()
+    elif publish_status == 'unreview':
```

Comment:
unreview, hide and so on are not really statuses but rather actions, aren't they? or you make the strings "unreviewed", "hidden" etc.